webpack.config.babel.js

```diff
@@ -18,61 +23,93 @@ const veryimportant = postcss.plugin("veryimportant", function() {
+    entry: process.env.NODE_ENV === "production" ? {
+        "tota11y": "./index.js",
+        "tota11y.min": "./index.js",
+    } : {
+        "tota11y": "./index.js",
     },
     output: {
-        path: path.join(__dirname, "build"),
-        filename: "tota11y.min.js",
+        path: path.join(__dirname, "dist"),
+        filename: "[name].js",
     },
     module: {
-        loaders: [
+        rules: [
             {
                 test: /\.js$/,
                 exclude: /node_modules/,
-                loader: "babel-loader",
-                query: {
-                    presets: ["env", "react"],
-                    plugins: [
-                        ["transform-react-jsx", {pragma: options.jsxPragma}]
-                    ],
-                },
+                use: [
+                    {
+                        loader: "babel-loader",
+                        options: {
+                            presets: [
+                                "env",
+                                "react",
+                            ],
+                            plugins: [
+                                ["transform-react-jsx", { pragma: options.jsxPragma }]
+                            ],
+                        },
+                    },
+                ],
+            },
+            {
+                test: /\.handlebars$/,
+                use: [
+                    {
+                        loader: "handlebars-loader",
+                    },
+                ],
             },
-            { test: /\.handlebars$/, loader: "handlebars", },
             {
                 test: /\.less$/,
-                loader: "style!css!postcss!less",
+                use: [
+                    "style-loader",
+                    { loader: "css-loader", options: { importLoaders: 1 } },
+                    {
+                        loader: "postcss-loader",
+                        options: {
+                            plugins: [
+                                veryimportant,
+                                autoprefixer({ browsers: ["> 1%"] }),
+                            ],
+                        }
+                    },
+                    "less-loader",
+                ],
             },
         ],
     },
-    plugins,
-    postcss: [
-        veryimportant,
-        autoprefixer({browsers: ["> 1%"]}),
+    plugins: [
+        new CleanWebpackPlugin(),
+        // Add a banner to our bundles with a version number, date, and
+        // license info
+        new webpack.BannerPlugin({
+            banner: bannerTemplate({
+                version: require("./package.json").version,
+                date: new Date().toISOString().slice(0, 10),
+            }),
+            entryOnly: true,
+        }),
+
+        // Make the JSX pragma function available everywhere without the need
+        // to use "require"
+        new webpack.ProvidePlugin({
+            [options.jsxPragma]: path.join(__dirname, "utils", "element"),
```

Comment:
I'm a little unclear exactly what this is doing, but this is what we were doing before so I guess it's alright.

Reply:
My understanding is that this allows us to use JSX without React. We inject our own calls to support creating elements so that when JSX is compiled to JS, it will work.